hd-wallet/plugin.lua

```diff
@@ -1,440 +1,418 @@
-function derive_new_child(parent_key, childIdx)
-    local data = ""
-    -- if index is greater than equal to first hardend key
-    if tonumber(childIdx) >= FIRST_HARDENED_CHILD then
-        data = parent_key.key
-    else
-        -- parent is private
-        -- data equal to public key of parent private
-        if parent_key.version == PRIVATE_WALLET_VERSION then
-            data = public_key_for_private_key(string.sub(parent_key.key, 3, 66))
-        else
-            -- key is public
-            -- data equal to parent key
-           data = parent_key.key
-        end
-    end
-    
-    -- concatenate index into data
-    local index_hex = num_2_hex(childIdx, 8)
-    data = data..index_hex
-
-    -- import chain-code as hmac key
-    -- sign data as from hmac key
-    local sobject = assert(Sobject.import { name = "hmac", obj_type = "HMAC", value = Blob.from_hex(parent_key.chain_code), transient = true })
-    local mac =  assert(sobject:mac { data = Blob.from_hex(data), alg = 'SHA512'}).digest
-    local hmac = mac:hex()
-
-    child_key = {
-        index = index_hex,
-        chain_code = string.sub(hmac, 65, 128),
-        depth = num_2_hex(tonumber(parent_key.depth + 1), 2)
-    }
-    if parent_key.version == PRIVATE_WALLET_VERSION then
-        child_key.version = PRIVATE_WALLET_VERSION 
-        fingerprint = hash_160(public_key_for_private_key(string.sub(parent_key.key, 3, 66))) 
-        child_key.fingerprint = string.sub(fingerprint, 1, 8)
-
-        -- appending 00 to make key size 33 bit
-        local a = BigNum.from_bytes_be(Blob.from_hex(string.sub(hmac, 1, 64)))
-        local b = BigNum.from_bytes_be(Blob.from_hex(parent_key.key))
-        a:add(b)
-        a:mod(BigNum.from_bytes_be(Blob.from_hex(N)))   
-        hex_key = a:to_bytes_be():hex()
-
-        if (string.len( hex_key ) < 66) then
-            local offset = string.rep("0", 32-string.len( hex_key ))
-            hex_key = offset..hex_key
-        end
-
-        child_key.key = "00"..tostring(hex_key)
-    else
-        child_key.version = PUBLIC_WALLET_VERSION
-        fingerprint = hash_160(parent_key.key)
-        child_key.fingerprint = string.sub(fingerprint, 1, 8)
-        keyBytes = public_key_for_private_key(string.sub(hmac, 1, 64))
-
-        local secP256K1 = EcGroup.from_name('SecP256K1')
-        local comp_key_1 = Blob.from_hex(keyBytes)
-        local pt_1 = secP256K1:point_from_binary(comp_key_1)
-        local x1 = pt_1:x()
-        local y1 = pt_1:y()
-        local comp_key_2 = Blob.from_hex(parent_key.key)
-        local pt = secP256K1:point_from_binary(comp_key_2)
-        local x2 = pt:x()
-        local y2 = pt:y()
-        local p1 = secP256K1:point_from_components(x1, y1)
-        local p2 = secP256K1:point_from_components(x2, y2)
-        local p3 = p1 + p2
-        
-        child_key.key = compress_public_key(p3:x():to_bytes_be():hex(), p3:y():to_bytes_be():hex())
+local function parse_path(path)
```

Comment:
This function returns an array with the indices for sequential derivation
`m/0/0H/42/42H --> [nil, 0, 2 ** 31, 42, 42 + 2 ** 31]`
(it was part of the original plugin).